pkg/controller/core/workload_controller.go

```diff
@@ -264,7 +266,8 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		reportWorkloadEvictedOnce := workload.WorkloadEvictionStateInc(&wl, kueue.WorkloadDeactivated, "")
 		updated = workload.ResetChecksOnEviction(&wl, r.clock.Now()) || updated
 		if updated {
-			if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock); err != nil {
+			err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock)
+			if client.IgnoreNotFound(err) != nil {
 				return ctrl.Result{}, fmt.Errorf("setting eviction: %w", err)
 			}
```

Comment:
```suggestion
			if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true, r.clock); err != nil {
				if apierrors.IsNotFound(err) {
					return ctrl.Restul{}, nil
				}
				return ctrl.Result{}, fmt.Errorf("setting eviction: %w", err)
			}
```

This could lead to an unexpected eviction count by Prometheus when the `err` is NotFound.